006-strimzi-ui.md

```diff
@@ -0,0 +1,120 @@
+
+## Proposal
+
+This proposal will describe a Kafka administration UI capability. It will describe example capabilities which could be offered, cover at a high level how the UI code itself could be implemented (with more low level detail to be provided as required), how this could be delivered, as well as possible future work and how this could integrate with Strimzi today.
+
+### UI implementation
+
+At a high level, I would propose a UI implemented as follows:
+
+- It would be a Javascript based UI (using [babel](https://babeljs.io/) to provide latest ECMAScript capabilities in a cross browser compatible manner), using the [React](https://reactjs.org/) framework
+- This UI is hosted and provided via an [Express](https://expressjs.com/) server
+- This UI is built in a modular/metadata driven manner - allowing for easy extensibility, modification, and dynamic behaviour at runtime
+- Uses a [GraphQL](https://graphql.org/) API
+- Should be fully translatable, and accessible
+
+I would suggest that the UI implementation to follow the [Model View Controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) (MVC) pattern, with clear separation between business logic, and the view logic that renders the state of the UI. This would enable/allow the view logic to be modified/swapped in/out as required, but keep the same business logic. I would suggest using the [Carbon](https://www.carbondesignsystem.com/) design system as the view layer for the contributed UI, given it is open source, is supported, and I can provide both the design and implementation using it. As an example, the following is a mock up of how a Topics page could look using the Carbon design system, which would allow a user to view, create, edit and delete topics in a given cluster:
+
+![Topics mock up](./006-topicsdesign.png)
+
+By maintaining and enforcing a thin view layer, this should not stop other view layers being designed, implemented and contributed in other frameworks, such as [PatternFly](https://www.patternfly.org/v4/) or [Material](https://material.io/) for example.
+
+I would also suggest using Express as a server for this UI. This is due to it's small footprint, modularity and available 3rd party modules, such as [helmet for http security](https://helmetjs.github.io/), [Passport.js for authentication and authorization support](http://www.passportjs.org/) or [the graphql and express-graphql modules for GraphQL server support](https://graphql.org/graphql-js/running-an-express-graphql-server/).
```

Comment:
> It might be interesting to list the licenses of the main dependencies and check their compatibility with out Apache Software License 2.0.

Absolutley - as it happens most I mention are either Apache 2 themselves, or similar (such as MIT). Any dependency added should be pedegree checked.